include/thread_pool/thread_pool.hpp

```diff
@@ -150,11 +152,11 @@ template <typename Task, template<typename> class Queue>
 template <typename Handler>
 inline void ThreadPoolImpl<Task, Queue>::post(Handler&& handler, bool block)
 {
-    while(true)
+    for(int i = (block)? m_workers.size()-1 : 0; ; --i)
```

Comment:
intent is not clear here, please comment the code

Reply:
The parameter description is provided at the function declaration. Have renamed unclear parameter name 'block' to 'to_any_queue'. Thanks.

Reply:
Implementation has been updated.